app/src/main/java/com/concordia/mcga/adapters/POISearchAdapter.java

```diff
@@ -0,0 +1,283 @@
+package com.concordia.mcga.adapters;
+
+import android.content.Context;
+import android.support.v7.widget.AppCompatImageView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseExpandableListAdapter;
+import android.widget.TextView;
+
+import com.concordia.mcga.activities.R;
+import com.concordia.mcga.models.Building;
+import com.concordia.mcga.models.Campus;
+
+import java.util.ArrayList;
+
+import de.hdodenhof.circleimageview.CircleImageView;
+
+/**
+ * This adapter searches through both campuses to find POIs (for now, buildings) that match a query.
+ */
+public class POISearchAdapter extends BaseExpandableListAdapter {
+    private Context context;
+
+    public static final int SGW_INDEX = 0;
+    public static final int LOYOLA_INDEX = 1;
+
+    private ArrayList<Building> sgwFilteredList;
+    private ArrayList<Building> loyolaFilteredList;
+
+    /**
+     * The constructor initializes the empty lists that store the currently queried POIs
+     * @param context View context to search
+     */
+    public POISearchAdapter(Context context) {
+        this.context = context;
+
+        this.sgwFilteredList = new ArrayList<>();
+        this.loyolaFilteredList = new ArrayList<>();
+    }
+
+    /**
+     * Evaluates the number of campuses active based on the current queried POIs
+     * @return The number of active campuses
+     */
+    @Override
+    public int getGroupCount() {
+        boolean sgwClear = getGroupIsEmpty(SGW_INDEX);
+        boolean loyClear = getGroupIsEmpty(LOYOLA_INDEX);
+
+        if (sgwClear && loyClear) {
+            return 0;
+        } else if (sgwClear || loyClear) {
+            return 1;
+        } else {
+            return 2;
+        }
+    }
+
+    /**
+     * This overrides the BaseExpandableList function to get the number of children (POI) entries
+     * @param i Campus index
+     * @return Number of POIs specific to the passed campus
+     */
+    @Override
+    public int getChildrenCount(int i) {
+        Campus camp = (Campus)getGroup(i);
+        if (camp == Campus.SGW) {
+            return sgwFilteredList.size();
+        } else if (camp == Campus.LOY) {
+            return loyolaFilteredList.size();
+        } else {
+            return 0;
+        }
+    }
+
+    /**
+     * This overrides the BaseExpandableList function to get the campus specified by an index.
+     * This function is written specifically to address the quirks in the way BaseExpandableList
+     * handles groups and children - it is used internally to get the "group" (campus) based on the
+     * number of visible groups, so we have to handle different cases depending on whether the
+     * POIs filtered by the query come from a single campus or both of them.
+     * @param i Active campus index
+     * @return The campus specified by the index
+     */
```

Comment:
We have static campuses, and BaseExpandableListAdapter assumes a dynamic list of groups. If this function were just called by the programmer it would be OK to just return Campus.SGW on the SGW_INDEX and vice versa, however Android calls this function at different times based on an index derived from getGroupCount.

Since we want to maintain the ordering of our static campuses while avoiding breaking the features, we have to account for the different cases depending on the groupCount - when it's zero, it should return null. If the groupcount is one, we have to find out which campus is the empty one. If the groupcount is two, we use the expected behaviour.